Lib/multiprocessing/queues.py

```diff
@@ -63,11 +63,15 @@ def __getstate__(self):
     def __setstate__(self, state):
         (self._ignore_epipe, self._maxsize, self._reader, self._writer,
          self._rlock, self._wlock, self._sem, self._opid) = state
-        self._after_fork()
+        self._notempty = threading.Condition(threading.Lock())
+        self._reset()
 
     def _after_fork(self):
         debug('Queue._after_fork()')
-        self._notempty = threading.Condition(threading.Lock())
+        self._notempty._at_fork_reinit()
+        self._reset()
+
+    def _reset(self):
```

Comment:
I would prefer to add a after_fork=False parameter. If after_fork is true, call _at_fork_reinit(). Otherwise, create a new lock.

Reply:
Good!